### ISSUE TITLE
(invalid-function lambda) when using `def-auto-persp'

### ISSUE THREAD

**Issue by terlar** [Bad-ptr/persp-mode.el#61]
Since the latest update there seems to have been some change introduced that breaks this, if I roll-back to the latest release version `2.9.4` there is no issue, but if I use the latest package version from MELPA I get a [backtrace](https://github.com/Bad-ptr/persp-mode.el/files/587212/backtrace.txt) during start.

My config block for auto-persp:
```
  (defvar after-find-file-hook nil)
  (advice-add 'find-file :after
    (lambda (&rest args) (run-hooks 'after-find-file-hook)))

  (def-auto-persp "projectile"
    :parameters '((dont-save-to-file . t))
    :hooks '(after-find-file-hook)
    :switch 'frame
    :predicate
    (lambda (_buffer)
      (when (and (buffer-file-name) (projectile-project-p))
        t))
    :get-name-expr
    (lambda ()
      (projectile-project-name)))
```

I have also tried to prefix the lambda with `#'` or `'`, but none of these works. Did something change in the latest version, is this not the way to do this anymore?

Thanks for a great package :+1:

**Comment by Bad-ptr** [comment]
Hmm. Really, this is a bug.
For now as a workaround you can try this:
```elisp
(def-auto-persp "projectile"
    :parameters '((dont-save-to-file . t))
    :hooks '(after-find-file-hook)
    :switch 'frame
    :predicate
    '((lambda (_buffer)
        (when (and (buffer-file-name) (projectile-project-p))
          t)))
    :get-name-expr
    (lambda ()
      (projectile-project-name)))
```

**Comment by Bad-ptr** [comment]
Must be fixed in d38f7546a070b1a4f8c1fcd617d00e3767d716b0
